src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -0,0 +1,122 @@
+                "fxr",
+                Path.GetFileName(dotNetCli.GreatestVersionHostFxrPath),
+                RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostfxr"));
+            result.Should().Pass()
+                .And.HaveStdOutContaining($"hostfxr_path: {expectedFxrPath}".ToLower());
+        }
+
+        [Fact]
+        public void GetHostFxrPath_WithAssemblyPath_AppLocalFxr()
+        {
+            TestProjectFixture fixture = sharedState.StandaloneAppFixture;
+            Command.Create(sharedState.NativeHostPath, $"{GetHostFxrPath} {fixture.TestProject.AppDll}")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdOutContaining($"hostfxr_path: {fixture.TestProject.HostFxrDll}".ToLower());
+        }
+
+        [Fact]
+        public void GetHostFxrPath_WithAssemblyPath_DotNetRootEnvironment()
+        {
+            string expectedFxrPath = Path.Combine(
+                dotNetCli.GreatestVersionHostFxrPath,
+                RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostfxr"));
+            Command.Create(sharedState.NativeHostPath, $"{GetHostFxrPath} {sharedState.NativeHostPath}")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .EnvironmentVariable("DOTNET_ROOT", dotNetCli.BinPath)
+                .Execute()
+                .Should().Pass()
+                .And.HaveStdOutContaining($"hostfxr_path: {expectedFxrPath}".ToLower());
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public RepoDirectoriesProvider RepoDirectories { get; }
+            public TestProjectFixture StandaloneAppFixture { get; }
+            public string NativeHostPath { get; }
+
+            private readonly string baseDir;
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+
+                baseDir = SharedFramework.CalculateUniqueTestDirectory(Path.Combine(TestArtifact.TestArtifactsPath, "nativeHosting"));
+                Directory.CreateDirectory(baseDir);
+
+                string nativeHostName = RuntimeInformationExtensions.GetExeFileNameForCurrentPlatform("nativehost");
+                NativeHostPath = Path.Combine(baseDir, nativeHostName);
+
+                // Copy over native host and nethost
+                string nethostName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("nethost");
+                File.Copy(Path.Combine(RepoDirectories.CorehostPackages, nethostName), Path.Combine(baseDir, nethostName));
+                File.Copy(Path.Combine(RepoDirectories.Artifacts, "corehost_test", nativeHostName), NativeHostPath);
+
+                var standaloneAppFixture = new TestProjectFixture("StandaloneApp", RepoDirectories);
+                StandaloneAppFixture = standaloneAppFixture
```

Comment:
I know this works, but we should try to avoid full blown build/publish unless necessary. It is really slow. For this test all we need is a fake `app.dll` and the `hostfxr` next to it.

On top of that, the `hostfxr` can also be a fake (empty file).

Let's try to make new tests as fast as possible - it already takes cca 15 minutes to run the tests, way too long.

Reply:
Switched to creating a fake `app.dll` / `hostfxr.dll` instead of using an actual project.